### PR TITLE
Added static analysis step to the CI.

### PR DESCRIPTION
**Pull request by rupertj**

### CODE REVIEW

**Comment by finnlewis** [comment]
Thanks for this @rupertj 

I understand from @willguv that we're aiming to bring localgov_publications into the default core localgov installation.

With that in mind, it would be good to be consistent in our testing across the repositories.

@millnut is currently working on setting up shared workflows, specifically with the aim of pulling in consistent workflows across the board, so it would be great to get input from @millnut on this pull request.

Also flagging for @stephen-cox too... should we be aiming to be consistent with  https://github.com/localgovdrupal/localgov_project/blob/3.x/.github/workflows/test.yml in the short term?

Might we want to add some of these changes to localgov_project?

**Comment by millnut** [comment]
Hi @rupertj if you could update this to use the [new shared workflow ](https://github.com/localgovdrupal/localgov_shared_workflows) that would be great. I've added an example to the [localgov/localgov_alert_banner](https://github.com/localgovdrupal/localgov_alert_banner/pull/324).

I know on Slack we discussed the possibility of modules having a higher PHPStan level vs the default level and I'm all for this, especially with the Drupal Core plans to increase this level in future.

What I think would be good is creating a PR at https://github.com/localgovdrupal/localgov_shared_workflows to add an optional input for the workflow, something like `use_module_phpstan_config` or similar which would allow you to use a module-level config.

**Review by stephen-cox — DISMISSED** [review]
As @millnut has mentioned, I think we're going to move all Github testing config to use https://github.com/localgovdrupal/localgov_shared_workflows/blob/1.x/.github/workflows/test-module.yml, so I suggest moving the change to the workflow there.

**Comment by rupertj** [comment]
Thanks all. I've moved this to use the new shared workflow, using the branch that I've made to it for this PR: https://github.com/localgovdrupal/localgov_shared_workflows/pull/1

**Comment by millnut** [comment]
Thanks @rupertj I was thinking about this change the other day, if it's only changing level maybe we just allow it to use the project phpstan but pass the level in the phpstan command e.g. `--level=5` and have a PHPSTAN_LEVEL optional input on the shared workflow. What do you think?

**Comment by rupertj** [comment]
@millnut The problem with just changing the level is going to be ignoring errors. IME as you turn up the error reporting level, you end up having to ignore more things. The errors you get at lower levels are unambiguous things that you just fix because they're obviously wrong. Some of the higher level stuff might be stuff you either can't fix or don't want to fix, so you have to add your own ignore rules for it.

I've had a look around d.o for examples, and the best one I've found is the one from search_api: https://git.drupalcode.org/project/search_api/-/blob/8.x-1.x/phpstan.neon?ref_type=heads

I did wonder if we could set the level and then use the baseline feature to let our modules suppress errors, but it seems it's impossible to set the baseline via an argument to phpstan. Baselines can only be set via the phpstan.neon. This also made me realise that if people want to use baselines, they'll need to have their own phpstan.neon too.

**Comment by millnut** [comment]
Hi @rupertj added a comment to https://github.com/localgovdrupal/localgov_shared_workflows/pull/1#issuecomment-2179093957 but I agree with your use cases mentioned

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -7,205 +7,12 @@ on:
...
-        run: docker-compose -f docker-compose.yml up -d
-
-      - name: Run PHPUnit tests
-        run: |
-          mkdir -p ./html/web/sites/simpletest && chmod 777 ./html/web/sites/simpletest
-          sed -i "s#http://localgov.lndo.site#http://drupal#" ./html/phpunit.xml.dist
-          docker exec -t drupal bash -c 'chown docker:docker -R /var/www/html'
-          docker exec -u docker -t drupal bash -c "cd /var/www/html && ./bin/phpunit /var/www/html/${{ env.LOCALGOV_DRUPAL_PROJECT_PATH }}"
+  tests:
+    uses: localgovdrupal/localgov_shared_workflows/.github/workflows/test-module.yml@feature/1.x/override-phpstan-neon-path
```
Comment (millnut) [inline .github/workflows/test.yml:14]: @ekes I think this needs to change to `localgovdrupal/localgov_shared_workflows/.github/workflows/test-module.yml@1.x` now it's merged on workflows.  @rupertj can you do a follow up PR to change this please

UPDATE: ignore the above it looks to be fix in this PR https://github.com/localgovdrupal/localgov_publications/pull/188

Comment (rupertj) [reply .github/workflows/test.yml:14]: Yup, I'm fixing all the things there. Will merge it in once I've got all the tests green again.

**Pull request merged**